Feature/sales coupon

Tim-M-S opened the pull request:
### All changes meet the following requirements
- [ ] Changelog entry was added
- [x] Changes have been tested by the author
- [x] Changes have been tested by the reviewer
- [x] Plugin can be built

@plentymarkets/ceres-io

Tim-M-S committed:
cleaned pr

Tim-M-S committed:
removed debug code and fixed typo

Tim-M-S committed:
changed button type from reset to button

Tim-M-S committed:
removed unused download method
added check for isLoading to submit() and finalize()

lks-mtzn commented on resources/js/src/app/components/item/EditCouponOverlay.vue:
```diff
@@ -0,0 +1,261 @@
+                        <div class="modal-body overflow-x-hidden modal-multi-row">
+                            <template v-for="coupon in couponData">
+                                <div class="row">
+                                    <div class="col-12 h5">Gutschein</div>
```
i18n

Tim-M-S replied:
done

lks-mtzn commented on resources/js/src/app/components/item/EditCouponOverlay.vue:
```diff
@@ -0,0 +1,261 @@
+<script>
+import NotificationService from "../../services/NotificationService";
+
+const ModalService  = require("../../services/ModalService");
```
import für beide nutzen

Tim-M-S replied:
done

lks-mtzn commented on resources/js/src/app/components/item/EditCouponOverlay.vue:
```diff
@@ -0,0 +1,261 @@
+
+        isPaid()
+        {
+            return this.paymentStatus === ("fullyPaid" || "overpaid");
```
Das funktioniert nicht wenn paymentStatus "overpaid" ist.

Tim-M-S replied:
done

lks-mtzn commented on resources/js/src/app/components/item/EditCouponOverlay.vue:
```diff
@@ -0,0 +1,261 @@
+
+    created()
+    {
+        for(let i=0; i < this.orderItem.quantity; i++)
```
Hier wird ein Eintrag zu viel drinne landen.

Tim-M-S replied:
😄👎

lks-mtzn commented on resources/js/src/app/components/item/EditCouponOverlay.vue:
```diff
@@ -0,0 +1,261 @@
+
+            this.isLoading = true;
+
+            ApiService.post("/rest/online_store/gift_card/generate_pdf", { orderId: this.orderItem.orderId, orderItemId: this.orderItem.id, accessKey: this.orderAccessKey}) // Route and Params missing
```
comment

Tim-M-S replied:
done

lks-mtzn commented on resources/js/src/base.js:
```diff
@@ -112,6 +112,7 @@ Vue.component("user-login-handler", UserLoginHandler);
 Vue.component("item-bundle", () => import("./app/components/item/ItemBundle.vue"));
 Vue.component("order-property-value", () => import("./app/components/item/OrderPropertyValue.vue"));
 Vue.component("quantity-input", () => import("./app/components/item/QuantityInput.vue"));
+Vue.component("edit-coupon-overlay", () => import("./app/components/item/EditCouponOverlay.vue"));
```
Wird die component wirklich überall gebraucht?

Tim-M-S replied:
moved to checkout.js

Tim-M-S committed:
added overlay for confirmation

Tim-M-S committed:
Merge branch 'feature/sales_coupon_finalization_notification' into feature/sales_coupon

# Conflicts:
#	resources/js/src/app/components/item/EditCouponOverlay.vue

fmutschler committed:
Update Template.properties

Tim-M-S committed:
fixed title,
removed debug comment,
added translations keys,
fixed check for isPaid(),
added closeConfirmModal() to finalize()

Tim-M-S committed:
added translations

Tim-M-S committed:
Merge branch 'feature/sales_coupon' of https://github.com/plentymarkets/plugin-ceres into feature/sales_coupon

# Conflicts:
#	resources/lang/de/Template.properties

fmutschler committed:
Update Template.properties

fmutschler committed:
Update Template.properties

Tim-M-S committed:
moved component to checkout.js
changes requested by lukas m.

stentrop committed:
Merge branch 'beta' into feature/sales_coupon

Tim-M-S committed:
submit closes correct modal

Tim-M-S committed:
centered positioning of modal removed
changed order of approve confirmation buttons

Tim-M-S committed:
separate loading state for finalization
modals always close after call

Tim-M-S committed:
changed spacing

Tim-M-S committed:
open pdf upon finalization

lks-mtzn committed:
Merge branch 'beta' into feature/sales_coupon

Tim-M-S committed:
Merge branch 'beta' into feature/sales_coupon
